src/components/ReviewList.tsx

```diff
@@ -0,0 +1,101 @@
+'use client';
+
+import React, { useEffect, useRef, useState } from 'react';
+
+import ReviewCard, { ReviewCardProps } from '@/components/card/ReviewCard';
+
+interface ReviewListProps {
+  data: {
+    total_pages: number;
+    size: number;
+    content: Omit<ReviewCardProps, 'onClick' | 'type'>[];
+    number: number;
+  };
+}
+
+const handleClick = (title: string) => {
+  alert(`Clicked on ${title}`);
+};
+
+export default function ReviewList({ data }: ReviewListProps) {
+  const [cardList, setCardList] = useState(
+    data.content.slice(0, data.size).map((item) => ({
+      ...item,
+      onClick: () => handleClick(item.title),
+      type: 'default' as const
+    }))
+  );
+  const [currentPage, setCurrentPage] = useState(0);
+  const [hasMore, setHasMore] = useState(currentPage < data.total_pages - 1);
+  const observerRef = useRef<HTMLDivElement | null>(null);
+
+  const fetchMoreData = () => {
+    if (!hasMore) return;
+
+    const newPage = currentPage + 1;
+    const start = newPage * data.size;
+    const end = start + data.size;
+    const newData = data.content.slice(start, end).map((item) => ({
+      ...item,
+      onClick: () => handleClick(item.title),
+      type: 'default' as const
+    }));
+
+    setCardList((prevData) => [...prevData, ...newData]);
+    setCurrentPage(newPage);
+
+    if (newPage >= data.total_pages - 1) {
+      setHasMore(false);
+    }
+  };
+
+  useEffect(() => {
+    const observer = new IntersectionObserver((entries) => {
+      if (entries[0].isIntersecting && hasMore) {
+        fetchMoreData();
+      }
+    });
+
+    if (observerRef.current) {
+      observer.observe(observerRef.current);
+    }
+
+    return () => {
+      if (observerRef.current) {
+        observer.unobserve(observerRef.current);
+      }
+    };
+  }, [hasMore]);
```

Comment:
useEffect의 의존성 배열에 fetchMoreData가 빠졌다고 나올 것 같습니다~
observer.disconect()는 감지기의 모든 기시성 변화 주시 대상을 해제합니다!

```suggestion
const newData = data.content.slice(start, end).map((item) => ({
      ...item,
      onClick: () => handleClick(item.title),
      type: 'default' as const
    }));

    setCardList((prevData) => [...prevData, ...newData]);
    setCurrentPage(newPage);

    if (newPage >= data.total_pages - 1) {
      setHasMore(false);
    }
  }, [currentPage, data.size, data.content, data.total_pages, hasMore]);

  useEffect(() => {
    const observer = new IntersectionObserver((entries) => {
      if (entries[0].isIntersecting && hasMore) {
        fetchMoreData();
      }
    });

    if (observerRef.current) {
      observer.observe(observerRef.current);
    }

    return () => {
      observer.disconnect();
    };
  }, [hasMore, fetchMoreData]);
```